projects/experimental/directives/pin/pin.directive.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Directive,
+    EventEmitter,
+    HostListener,
+    Inject,
+    Input,
+    Output,
+    ViewEncapsulation,
+} from '@angular/core';
+import {TUI_PLATFORM, TuiDirectiveStylesService, TuiPlatform} from '@taiga-ui/cdk';
+
+@Component({
+    host: {class: 'tui-pin'},
+    template: '',
+    styleUrls: ['./pin.styles.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+})
+export class TuiPinStyleComponent {}
```

Comment:
nit: since we do not have a pin component, only styles – there's no need to name it PinStyle, let's just move it to pin.component.ts and name it TuiPinComponent. That's how we did it everywhere, except for the buttons, because there's an actual button component there for icons and loader.

Reply:
fixed